test/Test.cpp

```diff
@@ -120,41 +176,50 @@ int main(int,char**)
 	java::lang::Object object = java::lang::Integer(23754);
 	if (jni::InstanceOf<java::lang::Number>(object) && jni::Cast<java::lang::Number>(object))
 	{
-		printf("%d\n", static_cast<int>(java::lang::Number(object)));
+		printf("%d\n", static_cast<jlong>(java::lang::Number(object)));
```

Comment:
It's not correct. You shouldn't use any of j* types, but rather C++ types. Try long, if that works on all platforms, then printf specifier should be %ld
https://www.cplusplus.com/reference/cstdio/printf/

Reply:
Fixed again

Reply:
Unfortunately, I can find many other places where jint is outputed as %d.

Reply:
check now